tools/site_cobble/gen_git_version_bsv.py

```diff
@@ -0,0 +1,36 @@
+import argparse
+
+from pathlib import Path
+from string import Template
+
+
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter
+)
+parser.add_argument(dest='version_code', help="Integer version code")
+parser.add_argument(dest='short_sha', help="8 character short SHA")
+parser.add_argument(dest='output_filename', help="string for bluespec package name")
+
+template = Template("""
+// Auto-generated as part of the FPGA build.
+package $package_name;
+
+Bit#(32) version = 'h$version;
+Bit#(32) sha = 'h$sha;
+
+function Bit#(8) byte_index(Bit#(32) value, Integer idx);
+    return value[8*idx + 7:8*idx];
+endfunction
+
+endpackage
```

Comment:
I think an earlier comment fell off, but I think getting an explicit "yes" or "no" is useful:

If this will often be used in a byte oriented manner I would advocate for exposing this as a `Vector#(4, Bit#(8))` instead of just a bit string:
```
import Vector::*;

Vector#(4, Bit#(8)) version = reverse(unpack('h$version));
Vector#(4, Bit#(8)) sha = reverse(unpack('h$sha));
```

You can then use the bytes as `version[0]`, `sha[1]`, etc. or the whole thing as bit string using `pack(sha)`.

Reply:
this is better than the function thing I did, fixed.